server/controllers/notification.js

```diff
@@ -0,0 +1,67 @@
+const mongoose = require('mongoose');
+const asyncHandler = require('express-async-handler');
+const Notification = require('../models/Notification');
+const Contest = require('../models/Contest');
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const senderId = req.user.id;
+  const { type } = req.body;
+  const typeList = ['submission', 'message'];
+  if (!typeList.includes(type)) {
+    res.status(400);
+    throw new Error('No such notification type.');
+  } else {
+    if (type === 'submission') {
+      const { files, _id: submissionId, contestId } = req.body;
+      // Use last file in submission array while submission featured photo is not yet set
+      const photo = files[files.length - 1];
+      const receiverId = await Contest.findById(contestId).then((contest) => {
+        return contest.user;
+      });
+      var params = { type, receiverId, senderId, contestId, submissionId, photo };
```

Comment:
Any reason we're using var here vs let/const ?

Reply:
I needed to pass `params` to line 27 which is outside the if-else block so I used `var`. Let me know if you think there's a better approach!

Reply:
You can declare variables using `let` outside of the if/else block and initialize them elsewhere as per your need

Reply:
Got it, thanks!